vscode/oso/syntaxes/polar.tmLanguage.json

```diff
@@ -80,7 +80,7 @@
     },
     "resource-block": {
       "name": "meta.resource-block",
-      "begin": "(resource|actor)\\s*([a-zA-Z][a-zA-Z0-9:]*)\\s*\\{",
+      "begin": "(resource|actor)\\s*([a-zA-Z][a-zA-Z0-9:-_]*)\\s*\\{",
```

Comment:
I don't think hyphen characters are permitted in Polar identifiers

Reply:
Here's where we permit underscores: https://github.com/osohq/oso/blob/69060d4b39f80ae722fcb25a2d6b8a13e150e885/polar-core/src/lexer.rs#L184

But we treat all hyphens as either [a unary/binary `-` operator](https://github.com/osohq/oso/blob/69060d4b39f80ae722fcb25a2d6b8a13e150e885/polar-core/src/lexer.rs#L473) or as [part of the exponent in exponential notation](https://github.com/osohq/oso/blob/69060d4b39f80ae722fcb25a2d6b8a13e150e885/polar-core/src/lexer.rs#L352).

Reply:
I believe you are correct. I didn't test it with hyphens, I assumed they were valid because I saw that same regex used later in the tmLanguage file.

Is it correct with the change I added?

Reply:
Yep, your change is correct, thank you! I suggested a few more improvements we can make while we're updating this syntax file in https://github.com/osohq/oso/pull/1462#discussion_r776123849. Gonna accept those changes and :shipit: 🎉